browser/main/NoteList/index.js

```diff
@@ -128,12 +132,56 @@ class NoteList extends React.Component {
   }
 
   componentDidUpdate (prevProps) {
-    const { location } = this.props
+    const { location, dispatch } = this.props
+    const { storage } = this.resolveTargetFolder()
     const { selectedNoteKeys } = this.state
     const visibleNoteKeys = this.notes.map(note => note.key)
     const note = this.notes[0]
     const prevKey = prevProps.location.query.key
     const noteKey = visibleNoteKeys.includes(prevKey) ? prevKey : note && note.key
+    const CSON = require('@rokt33r/season')
+
+    if (location.query.key !== prevKey) {
+      let focusNote = findNoteByKey(this.notes, location.query.key)
+
+      if (focusNote.filepath !== undefined) {
+        try {
+          this.fsWatcher.close()
+        } catch (e) {
```

Comment:
Please at least `console.error` the error. It might come in handy in the future.